variants/STM32MP157_DK/variant.cpp

```diff
@@ -254,6 +254,19 @@ void SystemClock_Config(void)
   /**Set the HSE division factor for RTC clock
   */
   __HAL_RCC_RTC_HSEDIV(24);
+
+  /* Configure ADCx clock prescaler */
+  PeriphClkInit.PeriphClockSelection = RCC_PERIPHCLK_ADC;
+  PeriphClkInit.AdcClockSelection = RCC_ADCCLKSOURCE_PER;
+  if (HAL_RCCEx_PeriphCLKConfig(&PeriphClkInit) != HAL_OK) {
+    Error_Handler();
+  }
+
+  /* Configure VREFBUF */
+  __HAL_RCC_VREF_CLK_ENABLE();
+  HAL_SYSCFG_VREFBUF_HighImpedanceConfig(SYSCFG_VREFBUF_HIGH_IMPEDANCE_DISABLE);
+  HAL_SYSCFG_EnableVREFBUF();
+
```

Comment:
Just in case, this block will be ignored since we do not use Engineering Boot mode.

Reply:
But in case we use Engineering Boot it will not be

Reply:
OK, it should be fine then. I just wanted to point out this block is not tested for this moment ;)

Reply:
It is good to put that  code under Engineering boot as this stuff is handled by Cortex A processor. that's why , you need in that case to well define ADC ressource in resource manager node in DT, so that OpenSTLinux does it. 
However, not familiar with Arduino distribution but not sure if SYSCFG/VREBUF config is relevant in SySTemclock Config. for sure , these kind of settings are also applicable for STM32 MCU families. In Cube example, this kind of code would be in HAL_PPP_msp_init (with PPP = ADC in our case)

Reply:
This could be in `HAL_ADC_msp_init`